site/cypress/e2e/Cypress-Feature/forums.spec.js

```diff
@@ -11,6 +11,41 @@ const reply3 = 'Cypress Reply 3 Cypress';
 const merged1 = 'Merged Thread Title: '.concat(title3, '\n\n', content3);
 const merged2 = 'Merged Thread Title: '.concat(title2, '\n\n', content2);
 const attachment1 = 'sea_animals.png';
+const isVisible = 'be.visible';
+const notVisible = 'be.not.visible';
+
+const goToForum = (user) => {
+    cy.login(user);
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-forum').click();
+};
+
+const uploadAttachmentAndDelete = (title) => {
+    cy.get('[data-testid="thread-list-item"]').contains(title).click();
+    cy.get('[data-testid="create-post-head"]').should('contain', title);
+    cy.get('[data-testid="edit-post-button"]').first().click();
+    cy.get('[data-testid="input-file1"]').selectFile('cypress/fixtures/sea_animals.png');
+    cy.get('[data-testid="file-upload-table-1"]').should('contain', 'sea_animals.png');
+    cy.get('[data-testid="forum-update-post"]').contains('Update Post').click();
+    cy.get('[data-testid="edit-post-button"]').first().click();
+    cy.get('[data-testid="mark-for-delete-btn"]').should('contain', 'Delete').click();
+    cy.get('[data-testid="mark-for-delete-btn"]').should('contain', 'Keep');
+    cy.get('[data-testid="forum-update-post"]').contains('Update Post').click();
+};
+
+const replyDisabled = (title, attachment) => {
+    cy.get('[data-testid="thread-list-item"]').contains(title).click();
+    // Reply button should be disabled by default with no text
+    cy.get('[value="Submit Reply to All"]').should('be.disabled');
```

Comment:
can you make it using data-testid? and on other place too

Reply:
That is beyond the scope of this PR, as the forum tests were already written with the selectors in place. The reason that they are green/red changes is because I moved them around. #10596 would be a better place to bring this up.

Reply:
Is there a reason why the functions were moved? It makes the actual changes slightly more difficult to spot.